cmd/kubelet/app/options/options.go

```diff
@@ -90,6 +90,16 @@ type KubeletFlags struct {
 	// additional configs to overwrite what is provided by default and in the KubeletConfigFile flag
 	KubeletDropinConfigDirectory string
 
+	// maxAllowableNUMANodes specifies the maximum number of NUMA Nodes that
```

Comment:
This should be moved to the kubelet config I think. We are aiming to avoid adding parameters to the kubelet CLI. I'd add it both in the staging directory and pkg/kubelet/apis/config.

Reply:
thanks for the review. 

> We are aiming to avoid adding parameters to the kubelet CLI.

yes, makes sense, I'll add it later.

Reply:
I also don't think this should be a top-level flag, but rather a topology-manager policy option. See the following for the PR that added policy options, including the very first option: https://github.com/kubernetes/kubernetes/pull/112914

Reply:
A policy option would indeed let us plug in the existing framework without new API reviews. Perhaps this is the way to go for all the new flags and features for topology manager?

Reply:
>  I also don't think this should be a top-level flag, but rather a topology-manager policy option. See the following for the PR that added policy options, including the very first option: 
https://github.com/kubernetes/kubernetes/pull/112914.

Interesting, I would have thought that this seems more apt as a top level kubelet flag behind a feature gate given its nature and relevance across all policies and policy options. If the value is not specified and feature gate not enabled, we can default to the old value of 8 and preserve backward compatibility with minimal code changes.

We would have to go through graduation process irrespective of its implementation as a top level flag or a policy option. What is the rationale behind proposing this as a policy option?

Reply:
So what's the conclusion here? We only need a policy option or a feature-gate. I prefer to have a policy option here.

Reply:
I still believe it should be a policy option. It scopes it beneath the TopologyManager better than having a top level flag.

Reply:
Based on your comments, I've made some updates, Could you retake a review? thanks!